pkg/agent/openflow/network_policy.go

```diff
@@ -1015,6 +1015,19 @@ func (c *client) GetPolicyFromConjunction(ruleID uint32) *v1beta1.NetworkPolicyR
 	return conjunction.npRef
 }
 
+func (c *client) GetPriorityFromConjunction(ruleID uint32) string {
+	conjunction := c.getPolicyRuleConjunction(ruleID)
+	priorities := conjunction.ActionFlowPriorities()
```

Comment:
Do we need lock?

Reply:
Does it mean that priority would be modified after we get this conjunction?

Reply:
Not just priority - ActionFlowPriorities() iterates conjunction.actionFlows. Probably @wenyingd and @Dyanngg can share some suggestions.

Reply:
I think we should be fine without acquiring a lock here. The lock for priority re-assignments is implemented per table at the reconciler level, which is hard to be obtained here. Even with the lock, we cannot ensure that the flow won't be reassigned a new priority during the time period where packetIn is processed and this function is called, which means the new reassigned flow priority will be returned anyways.

Reply:
My question is whether is safe to iterate actionFlows without a lock. Not just correctness.

Reply:
@Dyanngg : comments?

Reply:
Yes I believe it is safe. ActionFlowPriorities is a copy of current flow priorities.

Reply:
ActionFlowPriorities iterates c.actionFlows. Probably I should ask do we ever change c.actionFlows once it is initialized. As this is tricky (to access a field without locking), let us add least add some comments in ActionFlowPriorities to explain it is safe.

Reply:
Discussed with @jianjuns offline. Will add comments on why it's safe to access c.actionFlows in a follow-up PR.